### DISCUSSION TITLE
Static or Server-rendered or Client side rendered for a forum like website?

### DISCUSSION THREAD

**Discussion by qnxdev** [vercel/next.js#13856] [Help]
Which should I choose for a forum like page where 

the Author first posts something 
and there will be posts below that by anonymous
and posts are ranked by likes, 
the most liked post comes at the top.

I doubt if I go with Static the functions will be limited and each post might require page reload..
I don't want to compromise performance.

Thanks in advance

**Comment by enricoschaaf** [comment]
I would use getStaticProps() to generate the page when the Author first posts something with the help of the fallback prop, and then load the posts in on the client side. In the future you can use #11552 and specify a revalidate intervall and further down the road trigger a manual rebuild of the specific site whenever a user adds a new post or someone likes a post, but in every way you still need to fetch new data client side, in case there is a new post while a user has the site open. You could use something like Pusher Channels aka. Websockets to have a realtime forum or use SWR or React-Query with a refetch intervall.

**Reply by qnxdev** [reply]
Thanks.

But how do I implement dynamic routing like that in react where you can use render() to specify data which have to be used by the component?

for example if I need ` /apps/sr87hf8rh4f ` to load the page /apps/[appid].js with the data from sr87hf8rh4f object in the array.

Eg: 
`sr87hf8rh4f: {
       "name": "WhatsApp",
       "developer": "Facebook",
       "rating": 3.0
}`
